### PR TITLE
CHORE: Password page template

### PR DESCRIPTION
**Pull request by jamesbiggs**
## About these changes

Added a custom password page template

## How to check these changes

Make a page private with password view, open that page and you'll see the password page

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: sass/includes/_password-page.scss
Comment (krishan-torchbox) [inline sass/includes/_password-page.scss:3]: Do we need these empty styles are is it for future use?

File: templates/password_pages/password_required.html
Comment (ahosgood) [inline templates/password_pages/password_required.html:25]:
```suggestion
                    <div class="password-form__password">
```

File: templates/password_pages/password_required.html
Comment (ahosgood) [inline templates/password_pages/password_required.html:49]:
```suggestion
        function switchPasswordVisibility(event) {
            var x = document.getElementById("id_password");
            if (event.currentTarget.checked) {
                x.type = "password";
            } else {
                x.type = "text";
            }
        }
        
        var showHidePasswordCheckbox = document.createElement("input");
        showHidePasswordCheckbox.classList.add("password-form__show");
        showHidePasswordCheckbox.setAttribute("type", "checkbox");
        showHidePasswordCheckbox.addEventListener("change", switchPasswordVisibility);
        
        var formPasswordContainer = document.querySelector(".password-form__password");
        formPasswordContainer.appendChild(showHidePasswordCheckbox);
```

File: templates/password_pages/password_required.html
Comment (ahosgood) [inline templates/password_pages/password_required.html:29]:
```suggestion
```

File: templates/password_pages/password_required.html
Comment (ahosgood) [reply templates/password_pages/password_required.html:49]: This will (I hope) only add the show/hide password checkbox when JS is available rather than having it always available and do nothing when JS doesn't work or is disabled.

**Comment by ahosgood** [comment]
Looking at this now... 👀

**Comment by ahosgood** [comment]
Hope you don't mind @jamesbiggs but I made some styling a usability improvements. I'm going to approve and if you approve, I'll leave it to you to merge 😉

**Review by ahosgood — APPROVED** [review]

**Comment by jamesbiggs** [comment]
Cheers AJ, fine by me! This is just sort of a base template to work off, I think John said he'd be able to pick up the rest of the styling (we have a proper design for this on Figma)

**Comment by ahosgood** [comment]
Ah right - I shouldn't have assumed! 😅

**Pull request merged**
